src/modules/job-manager/update.c

```diff
@@ -127,14 +140,77 @@ update_request_create (struct update *update,
     return req;
 }
 
+static int expiration_from_duration (struct job *job,
+                                     flux_error_t *errp,
+                                     double duration,
+                                     double *expiration)
+{
+    if (duration == 0.)
+        *expiration = 0.;
+    else {
+        /*  Decode starttime of job's current R and add updated duration:
+        */
+        double starttime = -1.0;
+        if (!job->R_redacted
+            || json_unpack (job->R_redacted,
+                            "{s:{s:F}}",
+                            "execution",
+                            "starttime", &starttime) < 0
+            || starttime <= 0.) {
+            errprintf (errp, "unable to get job starttime");
+            return -1;
```

Comment:
No need to initialize `starttime` to -1 if it's not tagged as optional in the unpack.

However [RFC 20](https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_20.html#execution) does say it's optional...

Do we need to update the RFC to make it required?

Reply:
Note that jobs without a starttime will not be able to have their duration updated, so perhaps it does make sense to consider your suggested change to RFC 20. But the code will still operate correctly with the optional member unpack, so leaving it like that for now seems reasonable.